storage/innobase/lock/lock0lock.cc

```diff
@@ -5611,13 +5611,13 @@ dberr_t lock_trx_handle_wait(trx_t *trx)
   if (!trx->lock.wait_lock)
     return DB_SUCCESS;
   dberr_t err= DB_SUCCESS;
-  mysql_mutex_lock(&lock_sys.wait_mutex);
+  lock_sys.wait_mutex_lock();
```

Comment:
Spinning here might be useful. We are only holding some page latch. This would be exercised for `SKIP LOCKED` or when running `UPDATE WHERE non_matching_condition` at `READ UNCOMMITTED` or `READ COMMITTED` isolation level.